File: internal/models/support_ticket.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

// SupportTicket represents a support ticket
type SupportTicket struct {
	ID          uuid.UUID      `json:"id" gorm:"type:char(36);primaryKey"`
	UserID      uuid.UUID      `json:"user_id" gorm:"type:char(36);not null"`
	Subject     string         `json:"subject" gorm:"size:200;not null"`
	Description string         `json:"description" gorm:"type:text"`
	Status      string         `json:"status" gorm:"size:20;default:'open'"`
	Priority    string         `json:"priority" gorm:"size:20;default:'medium'"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`

	// Relationships
	User User `json:"user,omitempty" gorm:"foreignKey:UserID"`
}

// TableName specifies the table name for SupportTicket
func (SupportTicket) TableName() string {
	return "support_tickets"
}

// BeforeCreate will set a UUID rather than numeric ID
func (s *SupportTicket) BeforeCreate(tx *gorm.DB) error {
	if s.ID == uuid.Nil {
		s.ID = uuid.New()
	}
	return nil
}

File: src/models/support_ticket.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, Boolean
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from src.config.database import Base


class SupportTicket(Base):
    __tablename__ = "support_tickets"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    subject = Column(String(255), nullable=False)
    message = Column(Text, nullable=False)
    category = Column(String(50), default="general")  # account, transaction, security, technical, billing, general
    status = Column(String(50), default="open")  # open, in_progress, closed
    priority = Column(String(20), default="medium")  # low, medium, high, urgent
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    
    # Relationships
    user = relationship("User", back_populates="support_tickets")
    
    def __repr__(self):
        return f"<SupportTicket(id={self.id}, subject='{self.subject}', status='{self.status}')>"
